core/src/main/java/org/apache/iceberg/avro/AvroIO.java

```diff
@@ -178,7 +179,7 @@ static long findStartingRowPos(Supplier<SeekableInputStream> open, long start) {
           SYNC_READER.read(decoder, blockSync);
 
           if (!Arrays.equals(fileSync, blockSync)) {
-            throw new RuntimeIOException("Invalid sync at %s", nextSyncPos);
+            throw new FileHandlingException("Invalid sync at %s", nextSyncPos);
```

Comment:
We might be able to find a better exception for this.  I'm not clear on what this sync check is, but maybe there's something more appropriate we can use.

Reply:
Avro itself throws direct IOException for invalid sync cases as we can see here

https://github.com/apache/avro/blob/5548c9dff7a3bd851eacc791b52b21804f9f8408/lang/java/avro/src/main/java/org/apache/avro/file/DataFileStream.java#L330-L332

So, I too changed this now to use just IOException